crates/db_schema/src/impls/activity.rs

```diff
@@ -39,14 +40,15 @@ impl Crud for Activity {
 
 impl Activity {
   /// Returns true if the insert was successful
+  // TODO this should probably just be changed to an upsert on_conflict, rather than an error
```

Comment:
I dont think so. The inbox code already throws an error when activity with duplicate id is received. And for sending, we always generate a new, random id. So there must be a serious problem if we get a conflict here, better to catch that early.